src/tree/updater_quantile_hist.cc

```diff
@@ -102,33 +104,82 @@ void QuantileHistMaker::Builder::SyncHistograms(
   builder_monitor_.Start("SyncHistograms");
 
   const bool isDistributed = rabit::IsDistributed();
-
   const size_t nbins = hist_builder_.GetNumBins();
-  common::BlockedSpace2d space(nodes_for_explicit_hist_build_.size(), [&](size_t node) {
-    return nbins;
-  }, 1024);
+  if (!isDistributed) {
```

Comment:
This "if-else" is tooo large.
Need to split this in different methods or classes or apply another refactoring approach.

Reply:
refactored